src/theory/arith/arith_poly_norm.cpp

```diff
@@ -191,6 +211,75 @@ bool PolyNorm::isEqualMod(const PolyNorm& p, Rational& c) const
   return true;
 }
 
+Node PolyNorm::toNode(const TypeNode& tn) const
+{
+  std::vector<Node> sum;
+  NodeManager* nm = NodeManager::currentNM();
+  bool isArith = (tn.isInteger() || tn.isReal());
+  bool isBv = tn.isBitVector();
+  Kind multKind;
+  Node one;
+  if (isArith)
+  {
+    multKind = Kind::MULT;
+    one = nm->mkConstRealOrInt(tn, Rational(1));
+  }
+  else if (isBv)
+  {
+    multKind = Kind::BITVECTOR_MULT;
+    one = bv::utils::mkOne(tn.getBitVectorSize());
+  }
```

Comment:
Could we also include `addKind` and `zero`? Maybe even go HO and add a function `f` that creates constants...

Reply:
I'd prefer not adding zero here since it is possibly unnecessary node construction.